tests/test_python.py

```diff
@@ -96,9 +97,16 @@ def test_python_webserver_1(container_per_test):
     ).is_listening
 
     # start of the python http server
-    bci_pyt_serv = container_per_test.connection.run_expect(
+    container_per_test.connection.run_expect(
         [0], f"timeout 240s python3 -m http.server {port} &"
-    ).stdout
+    )
+
+    # wait for the webserver to start
+    timeout = 60
+    for i in range(timeout + 1):
+        assert i < timeout, "webserver startup timeout"
+        if container_per_test.connection.run("nc -z localhost {port}").rc:
```

Comment:
not sure if we need to use `f""` here, as this might consider `{port}` as a string and not a variable. Have you tried the code?

Reply:
also confusing, you are using `import time` but you don't use it. The way it's now, you need a `sleep(1)` somewhere, oder?

Reply:
Working on it right now.